docs/boot_modes.md

```diff
@@ -0,0 +1,111 @@
+# GardenLinux Boot Modes :potted_plant::penguin:
+
+GardenLinux supports two primary boot modes: Legacy and Unified System Image (USI).
+
+Legacy is the traditional way most Linux distros boot and also how all GardenLinux versions prior to 17?? used to boot.
+Legacy boot images are dual mode images, working both with UEFI boot and BIOS boot.
+
+USI mode is a novel approach, where the entire system gets embedded into a single UKI binary.
+This brings several advantages, such as allowing for in-place updates and enhanced security, but is only compatible with UEFI boot.
+
+|                          | USI                | Legacy             |
+|--------------------------|--------------------|--------------------|
+| UEFI boot                | :white_check_mark: | :white_check_mark: |
+| BIOS boot                | :x:                | :white_check_mark: |
+| secureboot / trustedboot | :heavy_check_mark: | :x:                |
+| in-place updates         | :white_check_mark: | :x:                |
+| TPM 2.0 disk encryption  | :heavy_check_mark: | :x:                |
+
+*( :heavy_check_mark: = optional )*
+
+> [!TIP]
+> If you are unsure which mode to pick: go for the *trustedboot* images, these are USIs with all optional security features enabled.
+
+## Legacy
+
+Legacy boot mode uses a standard partition layout.
+There is a boot partition, used both by syslinux for BIOS boot and systemd-boot for UEFI boot.
+The rootfs is on a simple ext4 partition, it is fully mutable.
+
+Legacy mode does not attempt to provide any protection against offline attacks.
+
+```shell
+# example disk layout for legacy mode
+
+vda                                                                         
+├─vda1
+│    vfat   FAT32 EFI   C1FE-FA19                                28M    67% /efi
+└─vda2
+     ext4   1.0   ROOT  aea6d56e-a2bf-944c-78fe-184a4d6add9f   29.1G     2% /
+```
+
+## Unified System Image (USI)
```

Comment:
so usi is a term that we invented, right?

Reply:
sort of, I've seen it used in a couple places before[^1], but it's not widely adopted yet.

[^1]: https://0pointer.net/blog/a-re-introduction-to-mkosi-a-tool-for-generating-os-images.html#:~:text=USIs%20(Unified%20System%20Images%20which%20are%20full%20OS%20images%20packed%20in%20a%20UKI)